### ISSUE TITLE
Latest native kernel issues with custom drivers

### ISSUE THREAD

**Issue by ihadeed** [clearlinux/distribution#1048]
**tl;dr**:
The latest kernel might have been built with a previous version of `gcc` and it causes issues when installing custom drivers.

---

Not sure if this is a real bug/issue or I'm just a newbie.

I have latest Clear Linux* _(30480)_ installed on my workstation and I recently installed two additional drivers:
- nvidia-430.34 (GeFore GTX 1080Ti)
- atlantic-2.2.6.0 (AQTION AQC 107)

Things were fine until I rebooted with the latest `kernel-native` update that was [released yesterday](https://github.com/clearlinux-pkgs/linux/releases/tag/5.2.2-802).

`dkms` failed to install the drivers mentioned above due to an incompatibility with the `gcc/plugins` version. I noticed that the Clear Linux* `gcc` package also was [updated yesterday](https://github.com/clearlinux-pkgs/gcc/releases/tag/9.1.1-587). 

I couldn't find any similar issues online so I decided to follow your [Kernel development guide](https://clearlinux.org/documentation/clear-linux/guides/maintenance/kernel-development) and build the latest kernel from source with the latest `gcc`.

That fixed my issues. After the kernel has successfully built, I rebooted into it and re-installed my drivers without any issues.

If this isn't a bug; I would appreciate some advice on what's the right way to handle this issue in the future without rebuilding the kernel.

**Comment by bryteise** [comment]
Are you using DKMS for your additional drivers?

**Comment by lebensterben** [comment]
>**tl;dr**: ...

@ihadeed 
Hi, I've also seen GCC errors when compilng dkms module for Nvidia driver. Can you take a look  #974 and check whether we have the same error message?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by miguelinux** [comment]
@ihadeed did you save the error output?

**Comment by ihadeed** [comment]
@bryteise Yes I'm using DKMS

@lebensterben No it's not the same error message

@miguelinux The logs got overwritten by the new installations. But it looked something like this:
```
# main error message
error: incompatible gcc/plugin versions

# file it was trying to compile/use
/lib/modules/5.2.2-802.native/build/scripts/gcc-plugins/structleak_plugin.so
```

**Comment by bwarden** [comment]
When we update gcc, we probably need to update all the kernels with -dev packages, too.

**Comment by Brandon2255p** [comment]
I am getting the same problem 

https://github.com/lwfinger/rtl8723bu/issues/148

**Comment by Brandon2255p** [comment]
@bwarden will you be rebuilding the kernel soon or should we follow what @ihadeed did and roll a custom build?

**Comment by lebensterben** [comment]
@bwarden I can reproduce this and the full build log is available [here](
https://gist.github.com/lebensterben/f0db88d4924a8ca929013213f3f64de7#file-make-log)

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by miguelinux** [comment]
Cl removes the GCC Plugins, so you will not have this issue.

The fix will be available in next public release (1 or 2 days).

**Comment by puneetse** [comment]
Confirmed working for me in 30550
